components/src/io_expander/stmpe1600/stmpe1600.ads

```diff
@@ -0,0 +1,178 @@
+--   (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE  --
+--   OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.   --
+--                                                                          --
+------------------------------------------------------------------------------
+
+with HAL.I2C;
+with HAL.GPIO;
+
+package STMPE1600 is
+
+   type STMPE1600_Expander (Port : not null HAL.I2C.Any_I2C_Port;
+                            Addr : HAL.I2C.I2C_Address) is limited private;
+   type Any_STMPE1600_Expander is access all STMPE1600_Expander;
+
+   type Pin_Polarity is (Low, High) with Size => 1;
+   type Pin_Direction is (Input, Output) with Size => 1;
+   type STMPE1600_Pin_Number is range 0 .. 15;
+
+   type STMPE1600_Pins is array (STMPE1600_Pin_Number) of Boolean
+     with Pack, Size => 16;
+
+   type STMPE1600_Pins_Direction is
+     array (STMPE1600_Pin_Number) of Pin_Direction
+     with Pack, Size => 16;
+
+   procedure Configure
+     (This   : in out STMPE1600_Expander;
+      Status : out Boolean);
+
+   procedure Set_Interrupt_Enable
+     (This     : in out STMPE1600_Expander;
+      Enable   : Boolean;
+      Polarity : Pin_Polarity;
+      Status   : out Boolean);
+   --  Enables/Disables the interrupt to the host.
+   --  Allows also to set whether the interrupt is active High or Low.
+
+   procedure Set_Interrupt_Mask
+     (This : STMPE1600_Expander;
+      Mask : STMPE1600_Pins;
+      Status : out Boolean);
+   --  Enables the interrupt from a particular GPIO interrupt source to the
+   --  host.
+
+   function Interrupt_Mask
+     (This : STMPE1600_Expander) return STMPE1600_Pins;
+   --  Returns the current value of the interrupt mask.
+
+   function Interrupt_State
+     (This : STMPE1600_Expander) return STMPE1600_Pins;
+   --  Tells if pins have changed since last query. Reading this value
+   --  clears the state.
+   --  The state is set regardless of the interrupt mask.
+
+   function Pins_State
+     (This : in out STMPE1600_Expander) return STMPE1600_Pins;
+
+   function Pin_State
+     (This : in out STMPE1600_Expander;
+      Pin  : STMPE1600_Pin_Number) return Boolean;
+
+   procedure Set_Pins_State
+     (This : in out STMPE1600_Expander;
+      Pins : STMPE1600_Pins);
+
+   procedure Set_Pin_State
+     (This  : in out STMPE1600_Expander;
+      Pin   : STMPE1600_Pin_Number;
+      State : Boolean);
+
+   procedure Set_Pins_Direction
+     (This      : STMPE1600_Expander;
+      Pins      : STMPE1600_Pins_Direction);
+
+   procedure Set_Pin_Direction
+     (This      : STMPE1600_Expander;
+      Pin       : STMPE1600_Pin_Number;
+      Direction : Pin_Direction);
+
+   procedure Set_Pin_Polarity_Inversion
+     (This            : STMPE1600_Expander;
+      Pin             : STMPE1600_Pin_Number;
+      Inversion_State : Boolean);
+   --  Enables or disables the polarity inversion of the given input pin
+
+   type STMPE1600_Pin
```

Comment:
I don't understand why making this type private? This makes the use of the HAL.GPIO interface useless (e.g. the ability to pass pins to a specific component, without having the explicit board definition).

In my case, this would break the implementation of the VL53L0x driver for example, as it needs access to the reset pin of the sensor, and I don't want to have it depending on a specific gpio expander.